quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -223,131 +227,135 @@ async fn index_metadata(
 /// Extends an existing SQL string with the generated filter range appended to the query.
 ///
 /// This method is **not** SQL injection proof and should not be used with user-defined values.
-fn write_sql_filter<V: Display>(
-    sql: &mut String,
-    field_name: impl Display,
-    filter_range: &FilterRange<V>,
-    value_formatter: impl Fn(&V) -> String,
-) {
-    match &filter_range.start {
-        Bound::Included(value) => {
-            let _ = write!(sql, " AND {} >= {}", field_name, (value_formatter)(value));
-        }
-        Bound::Excluded(value) => {
-            let _ = write!(sql, " AND {} > {}", field_name, (value_formatter)(value));
-        }
-        Bound::Unbounded => {}
+fn write_sql_filter<'a, V: Display>(
```

Comment:
The combination of passing the base statement as a mutable reference AND having to return the reference is weird. We should either pass the base statement as a mutable reference and return nothing OR pass the base statement by value and return it. I have a preference for the former because it also allows removing the lifetime.